
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="shortcut icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-6.1.5">
    
    
      
        <title>MUFFIN</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.21aed14c.min.css">
      
        
        <link rel="stylesheet" href="assets/stylesheets/palette.196e0c26.min.css">
        
          
          
          <meta name="theme-color" content="#7e56c2">
        
      
    
    
    
      
        
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto",-apple-system,BlinkMacSystemFont,Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono",SFMono-Regular,Consolas,Menlo,monospace}</style>
      
    
    
    
    
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="deep-purple" data-md-color-accent="deep-orange">
      
  
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#muffin" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid" aria-label="Header">
    <a href="." title="MUFFIN" class="md-header-nav__button md-logo" aria-label="MUFFIN">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header-nav__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header-nav__title" data-md-component="header-title">
      
        <div class="md-header-nav__ellipsis">
          <span class="md-header-nav__topic md-ellipsis">
            MUFFIN
          </span>
          <span class="md-header-nav__topic md-ellipsis">
            
              Home
            
          </span>
        </div>
      
    </div>
    
      <label class="md-header-nav__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0116 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 019.5 16 6.5 6.5 0 013 9.5 6.5 6.5 0 019.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" data-md-component="search-reset" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          

<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
    
      
    
    <a href="." class="md-tabs__link md-tabs__link--active">
      Home
    </a>
  </li>

      
    </ul>
  </div>
</nav>
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="MUFFIN" class="md-nav__button md-logo" aria-label="MUFFIN">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 003-3 3 3 0 00-3-3 3 3 0 00-3 3 3 3 0 003 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    MUFFIN
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Home
        <span class="md-nav__icon md-icon"></span>
      </label>
    
    <a href="." class="md-nav__link md-nav__link--active">
      Home
    </a>
    
      
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#index" class="md-nav__link">
    INDEX
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#figure" class="md-nav__link">
    Figure
  </a>
  
    <nav class="md-nav" aria-label="Figure">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-workflow" class="md-nav__link">
    The Workflow
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-parser-output" class="md-nav__link">
    The parser output
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    Installation
  </a>
  
    <nav class="md-nav" aria-label="Installation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#base-installation" class="md-nav__link">
    base installation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-conda-usage" class="md-nav__link">
    For conda usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-gcloud-usage" class="md-nav__link">
    For gcloud usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-containers-usage" class="md-nav__link">
    For containers usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-usage-of-software-installed-locally" class="md-nav__link">
    For usage of software installed locally
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#test-the-pipeline" class="md-nav__link">
    Test the pipeline
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    Usage
  </a>
  
    <nav class="md-nav" aria-label="Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-usage" class="md-nav__link">
    Basic usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-usage" class="md-nav__link">
    Advanced usage
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting" class="md-nav__link">
    Troubleshooting
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#options" class="md-nav__link">
    Options
  </a>
  
    <nav class="md-nav" aria-label="Options">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#-cpus" class="md-nav__link">
    --cpus
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-mem" class="md-nav__link">
    --mem
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-assembler" class="md-nav__link">
    --assembler
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-illumina" class="md-nav__link">
    --illumina
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-nanopore" class="md-nav__link">
    --nanopore
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-output" class="md-nav__link">
    --output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-profile" class="md-nav__link">
    -profile
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#complete-help-and-options" class="md-nav__link">
    Complete help and options
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bibliography" class="md-nav__link">
    BIBLIOGRAPHY
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#license" class="md-nav__link">
    License
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#contributing" class="md-nav__link">
    Contributing
  </a>
  
</li>
      
    </ul>
  
</nav>
    
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#index" class="md-nav__link">
    INDEX
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#figure" class="md-nav__link">
    Figure
  </a>
  
    <nav class="md-nav" aria-label="Figure">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-workflow" class="md-nav__link">
    The Workflow
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-parser-output" class="md-nav__link">
    The parser output
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#installation" class="md-nav__link">
    Installation
  </a>
  
    <nav class="md-nav" aria-label="Installation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#base-installation" class="md-nav__link">
    base installation
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-conda-usage" class="md-nav__link">
    For conda usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-gcloud-usage" class="md-nav__link">
    For gcloud usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-containers-usage" class="md-nav__link">
    For containers usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#for-usage-of-software-installed-locally" class="md-nav__link">
    For usage of software installed locally
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#test-the-pipeline" class="md-nav__link">
    Test the pipeline
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#usage" class="md-nav__link">
    Usage
  </a>
  
    <nav class="md-nav" aria-label="Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#basic-usage" class="md-nav__link">
    Basic usage
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#advanced-usage" class="md-nav__link">
    Advanced usage
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#troubleshooting" class="md-nav__link">
    Troubleshooting
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#options" class="md-nav__link">
    Options
  </a>
  
    <nav class="md-nav" aria-label="Options">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#-cpus" class="md-nav__link">
    --cpus
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-mem" class="md-nav__link">
    --mem
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-assembler" class="md-nav__link">
    --assembler
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-illumina" class="md-nav__link">
    --illumina
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-nanopore" class="md-nav__link">
    --nanopore
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-output" class="md-nav__link">
    --output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#-profile" class="md-nav__link">
    -profile
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#complete-help-and-options" class="md-nav__link">
    Complete help and options
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bibliography" class="md-nav__link">
    BIBLIOGRAPHY
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#license" class="md-nav__link">
    License
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#contributing" class="md-nav__link">
    Contributing
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="muffin">MUFFIN</h1>
<p>MUFFIN is a hybrid assembly and differential binning workflow for metagenomics, transcriptomics and pathway analysis.</p>
<p>A preprint is available here https://www.biorxiv.org/content/10.1101/2020.02.08.939843v1</p>
<h2 id="index">INDEX</h2>
<ol>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#figure">Figure</a> :<ul>
<li><a href="#the-workflow">Workflow</a></li>
<li><a href="#the-parser-output">Parser output</a></li>
</ul>
</li>
<li><a href="#installation">Installation</a> :<ul>
<li><a href="#base-installation">base installation</a></li>
<li><a href="#for-conda-usage">conda usage</a></li>
<li><a href="#for-gcloud-usage">gcloud usage</a></li>
<li><a href="#for-containers-usage">containers usage</a></li>
<li><a href="#for-usage-of-software-installed-locally">software installe locally</a></li>
</ul>
</li>
<li><a href="#test-the-pipeline">Test the pipeline</a></li>
<li><a href="#usage">Usage</a> :<ul>
<li><a href="#basic-usage">Basic usage</a></li>
<li><a href="#advanced-usage">Advanced usage</a></li>
</ul>
</li>
<li><a href="#troubleshooting">Troubleshooting</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#complete-help-and-options">Complete help and options</a></li>
<li><a href="#bibliography">Bibliography</a></li>
<li><a href="#license">License</a></li>
</ol>
<h2 id="introduction">Introduction</h2>
<p>MUFFIN aims at being a reproducible pipeline for metagenome assembly
of crossed illumina and nanopore reads.</p>
<p>MUFFIN uses the following software </p>
<table>
<thead>
<tr>
<th>Task</th>
<th>Software</th>
<th>Version</th>
<th>Docker</th>
<th>Image version</th>
</tr>
</thead>
<tbody>
<tr>
<td>QC illumina</td>
<td><a href="https://github.com/OpenGene/fastp">fastp</a></td>
<td>0.20.0</td>
<td><a href="https://hub.docker.com/r/nanozoo/fastp">LINK</a></td>
<td>0.20.0--78a7c63</td>
</tr>
<tr>
<td>QC ont</td>
<td>automated way to discard shortest reads</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td><a href="https://github.com/rrwick/Filtlong">filtlong</a></td>
<td>0.2.0</td>
<td><a href="https://hub.docker.com/r/nanozoo/filtlong">LINK</a></td>
<td>v0.2.0--afa175e</td>
</tr>
<tr>
<td>metagenomic composition of ont</td>
<td><a href="https://sourmash.readthedocs.io/en/latest/">sourmash</a></td>
<td>2.0.1</td>
<td><a href="https://hub.docker.com/r/nanozoo/sourmash">LINK</a></td>
<td>2.0.1--6970ddc</td>
</tr>
<tr>
<td>Hybrid assembly</td>
<td><a href="http://cab.spbu.ru/software/spades/">Meta-spades</a></td>
<td>3.13.1</td>
<td><a href="https://hub.docker.com/r/nanozoo/spades">LINK</a></td>
<td>3.13.1--2c2a4c0</td>
</tr>
<tr>
<td></td>
<td><a href="https://github.com/rrwick/Unicycler">unicycler</a></td>
<td>0.4.7</td>
<td><a href="https://hub.docker.com/r/nanozoo/unicycler">LINK</a></td>
<td>0.4.7-0--c0404e6</td>
</tr>
<tr>
<td>Long read assembly</td>
<td><a href="https://github.com/fenderglass/Flye">MetaFlye</a></td>
<td>2.7</td>
<td><a href="https://hub.docker.com/r/nanozoo/flye">LINK</a></td>
<td>2.7--957a1a1</td>
</tr>
<tr>
<td>polishing</td>
<td><a href="https://github.com/lbcb-sci/racon">racon</a></td>
<td>1.4.13</td>
<td><a href="https://hub.docker.com/r/nanozoo/racon">LINK</a></td>
<td>1.4.13--bb8a908</td>
</tr>
<tr>
<td></td>
<td><a href="https://github.com/nanoporetech/medaka">medaka</a></td>
<td>1.0.3</td>
<td><a href="https://hub.docker.com/r/nanozoo/medaka">LINK</a></td>
<td>1.0.3--7c62d67</td>
</tr>
<tr>
<td></td>
<td><a href="https://github.com/broadinstitute/pilon/wiki">pilon</a></td>
<td>1.23</td>
<td><a href="https://hub.docker.com/r/nanozoo/pilon">LINK</a></td>
<td>1.23--b21026d</td>
</tr>
<tr>
<td>mapping</td>
<td><a href="https://github.com/lh3/minimap2">minimap2</a></td>
<td>2.17</td>
<td><a href="https://hub.docker.com/r/nanozoo/minimap2">LINK</a></td>
<td>2.17--caba7af</td>
</tr>
<tr>
<td></td>
<td><a href="http://bio-bwa.sourceforge.net/">bwa</a></td>
<td>0.7.17</td>
<td><a href="https://hub.docker.com/r/nanozoo/pilon">LINK</a></td>
<td>1.23--b21026d</td>
</tr>
<tr>
<td></td>
<td><a href="http://www.htslib.org/">samtools</a></td>
<td>1.9</td>
<td><a href="https://hub.docker.com/r/nanozoo/minimap2">LINK</a></td>
<td>2.17--caba7af</td>
</tr>
<tr>
<td>retrieve reads mapped to contig</td>
<td><a href="https://github.com/lh3/seqtk">seqtk</a></td>
<td>1.3</td>
<td><a href="https://hub.docker.com/r/nanozoo/seqtk">LINK</a></td>
<td>1.3--dc0d16b</td>
</tr>
<tr>
<td>Binning</td>
<td><a href="https://bitbucket.org/berkeleylab/metabat/src/master/">Metabat2</a></td>
<td>2.13</td>
<td><a href="https://hub.docker.com/r/nanozoo/metabat2">LINK</a></td>
<td>2.13--0e2577e</td>
</tr>
<tr>
<td></td>
<td><a href="https://sourceforge.net/projects/maxbin2/">maxbin2</a></td>
<td>2.2.7</td>
<td><a href="https://hub.docker.com/r/nanozoo/maxbin2">LINK</a></td>
<td>2.2.7--b643a6b</td>
</tr>
<tr>
<td></td>
<td><a href="https://github.com/BinPro/CONCOCT">concoct</a></td>
<td>1.1.0</td>
<td><a href="https://hub.docker.com/r/nanozoo/concoct">LINK</a></td>
<td>1.1.0--03a3888</td>
</tr>
<tr>
<td></td>
<td><a href="https://github.com/bxlab/metaWRAP">metawrap</a></td>
<td>1.2.2</td>
<td><a href="https://hub.docker.com/r/nanozoo/metawrap">LINK</a></td>
<td>1.2.2--de94241</td>
</tr>
<tr>
<td>qc binning</td>
<td><a href="https://ecogenomics.github.io/CheckM/">checkm</a></td>
<td>1.0.13</td>
<td><a href="https://hub.docker.com/r/nanozoo/nanoplot">LINK</a></td>
<td>1.0.13--248242f</td>
</tr>
<tr>
<td>Taxonomic Classification</td>
<td><a href="https://sourmash.readthedocs.io/en/latest/">sourmash</a> using the <a href="https://gtdb.ecogenomic.org/">gt-DataBase</a></td>
<td>2.0.1</td>
<td><a href="https://hub.docker.com/r/nanozoo/sourmash">LINK</a></td>
<td>2.0.1--6970ddc</td>
</tr>
<tr>
<td></td>
<td><a href="https://gtdb.ecogenomic.org/">GTDB</a></td>
<td>version r89</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Annotations (bin and RNA)</td>
<td><a href="https://github.com/eggnogdb/eggnog-mapper/wiki/eggNOG-mapper-v2">eggNOG</a></td>
<td>2.0.1</td>
<td><a href="https://hub.docker.com/r/nanozoo/eggnog-mapper">LINK</a></td>
<td>2.0.1--d5e0c8c</td>
</tr>
<tr>
<td></td>
<td><a href="http://eggnog5.embl.de/#/app/home">eggNOG DB</a></td>
<td>v5.0</td>
<td></td>
<td></td>
</tr>
<tr>
<td><em>De novo</em> transcript and quantification</td>
<td><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Trinity</a></td>
<td>2.9.1</td>
<td><a href="https://hub.docker.com/r/nanozoo/trinity">LINK</a></td>
<td>2.9.1--82fe26c</td>
</tr>
<tr>
<td></td>
<td><a href="https://github.com/COMBINE-lab/salmon">Salmon</a></td>
<td>0.15.0</td>
<td><a href="https://hub.docker.com/r/nanozoo/trinity">LINK</a></td>
<td>2.9.1--82fe26c</td>
</tr>
</tbody>
</table>
<h2 id="figure">Figure</h2>
<h3 id="the-workflow">The Workflow</h3>
<p><img alt="MUFFIN FLOWCHART FIGURE" src="figures/Muffin_Workflow_simple.png" /></p>
<h3 id="the-parser-output">The parser output</h3>
<p><img alt="PARSER OUTPUT FIGURE" src="figures/PANKEGG_simple.png" /></p>
<h2 id="installation">Installation</h2>
<h3 id="base-installation">base installation</h3>
<p>You need to install nextflow Version 20.07+ ( https://www.nextflow.io/ )</p>
<pre><code class="language-sh"># verify Java version (at least version 8+)
java -version 

# Setup nextflow (it will create a nextflow executable file in the current directory)
curl -s https://get.nextflow.io | bash

# If you want the pipeline installed locally use the following
git clone https://github.com/RVanDamme/MUFFIN.git

# If you want to not install the pipeline use the following when running nextflow

nextflow run  RVanDamme/MUFFIN --parameters.....

</code></pre>
<h3 id="for-conda-usage">For conda usage</h3>
<p>If you use conda, you don't need extra installations.
An error might occur with the installation of metawrap, if so please consult <a href="#troubleshooting">Troubleshooting</a>.</p>
<h3 id="for-gcloud-usage">For gcloud usage</h3>
<p>If you use the google lifescience ressources you first need to setup a few parameters.</p>
<p>In the nextflow.config you need to change the parameters of gcloud to correspond to your project (line 67 to 78).</p>
<pre><code class="language-sh">    gcloud {  
        //workDir = &quot;/tmp/nextflow-docker_pipelines-$USER&quot;
        process.executor = 'google-lifesciences'
        process.memory = params.memory
        bucketDir = 'gs://bucket/work-dir' // change this to your bucket where you want the workfile to be stored
        google { project = 'project-name-111111'; zone = 'europe-north1-a' } // insert your project ID as well as the zone(s) you want to use
        // you can also use {region = 'europe-north1'} instead of zone
        google.lifeSciences.copyImage = 'google/cloud-sdk:latest'
        google.lifeSciences.preemptible = true
        google.lifeSciences.bootDiskSize = &quot;10GB&quot;
        google.lifeSciences.debug = true
        //includeConfig 'configs/preemptible.config'
    }
</code></pre>
<p>You will also have to change the bucket to store the different database in:
    /modules/checkmgetdatabases.nf ; /modules/eggnog_get_databases.nf ; /modules/sourmashgetdatabase.nf</p>
<p>To do so just edit the line using your bucket. keep the structure for more clarity (e.g. keep the "/databases-nextflow/sourmash" part).</p>
<p>Example:</p>
<pre><code class="language-sh">if (workflow.profile.contains('gcloud')) {publishDir 'gs://gcloud_storage/databases-nextflow/sourmash', mode: 'copy', pattern: &quot;genbank-k31.lca.json.gz&quot; }
#becomes
if (workflow.profile.contains('gcloud')) {publishDir 'gs://MY_STORAGE/databases-nextflow/sourmash', mode: 'copy', pattern: &quot;genbank-k31.lca.json.gz&quot; }
</code></pre>
<p>If you desire run on gcloud without the preemptible parameter activated just edit the line 74 of nextflow.config to false.</p>
<h3 id="for-containers-usage">For containers usage</h3>
<p>If you use containers either docker or singularity, you don't need extra installations.</p>
<h3 id="for-usage-of-software-installed-locally">For usage of software installed locally</h3>
<p>You just need to have all the software used in the pipeline (see table above) installed and in your $PATH.</p>
<h2 id="test-the-pipeline">Test the pipeline</h2>
<p>To test the pipeline we have a subset of 5 bins available at https://osf.io/9xmh4/
A detailed explanation of all the parameter is available in <a href="#usage">Usage</a>, the most important for the test is the profile executor and engine.
To run it you just need to add "test" in the -profile parameter e.g.:</p>
<pre><code>#test locally with conda, you need to specify cpus and ram available
nextflow run RVanDamme/MUFFIN --output results_dir  --cpus 8 --memory 32g -profile local,conda,test

#test locally with docker, you can change the cpus and ram in configs/containers.config
# this test also run the transcriptomics analysis with --rna
nextflow run RVanDamme/MUFFIN --output results_dir --rna -profile local,docker,test

#test using gcloud with docker, you can change the cpus and ram in configs/containers.config
# this test use flye instead of spades with the --assembler metaflye
nextflow run RVanDamme/MUFFIN --output results_dir --assembler metaflye -profile gcloud,docker,test
</code></pre>
<p>The subset contains also RNA data to test with transcriptomics analysis you just need to activate it using "--rna"
The results of the different test run are available at https://osf.io/m5czv/</p>
<h2 id="usage">Usage</h2>
<h3 id="basic-usage">Basic usage</h3>
<pre><code>path/to/nextflow run $MUFFIN_pipeline --output results_dir --assembler $assembler --illumina fastq_ill/ --ont fastq_ont/ --cpus 16 --memory 64g --modular full -profile $profile_executor,$profile_engine
</code></pre>
<p>$MUFFIN_pipeline is either "path/to/MUFFIN/main.nf" or "RVanDamme/MUFFIN"</p>
<p>$assembler is either:
  - "metaspades" for hybrid assembly
  - "metaflye" for long-read assembly with short-reads polishing</p>
<p>$profile_executor can be:
  - "local" to run on your computer
  - "gcloud" to run on google life science cloud computing (you need to setup your project in nextflow.config)
  - "slurm" to run on HPC using slurm (e.g. UPPMAX)</p>
<p>$profile_engine can be:
  - "local_engine" to execute the software installed locally
  - "conda" to execute using conda installation
  - "docker" and "singularity" to execute using the docker container</p>
<p>You can add "-resume" at the end of the command to restart it while keeping the process that succeeded
This is often used in case or error in the pipeline or if you modify slightly the command and want to avoid running everything again.
One exemple is to run the pipeline without RNA and rerun it adding RNA data, in this specific case the second time you add:</p>
<pre><code>--rna path/to/rna -resume
</code></pre>
<p>Only the transcript processes and final parsing will be run</p>
<h3 id="advanced-usage">Advanced usage</h3>
<p>You can use RNA data to have transcriptomics analysis to do so add "--rna path/to/fastq_rna/"</p>
<p>You can run only partially the pipeline to do so change --modular to the right parameter:
 - "full" run the 3 steps of MUFFIN (assemble, classify, annotate)
 - "assemble" run the assembly and binning
 - "classify" run the classification of the bins (require a different input)
 - "annotate" run the annotation step of the bins (require a different input) and RNA if provided
 - "assemb-class" run assemble and classify step
 - "assemb-annot" run assemble and classify step
 - "class-annot" run classify and annotate step (require a different input)</p>
<p>To run classify and annotate independently from the assemble you need to provide a CSV file of the bins
The structure of the file should correspond to:</p>
<pre><code>Samplename,path/to/bin1.fa
Samplename,path/to/bin2.fa
...
Samplename,path/to/binX.fa
Samplename,path/to/binY.fa
</code></pre>
<p>If you run "classify" with or without "annotation" use "--bin_classify"
If you run "annotate" without "classify" use "--bin_annotate"</p>
<h2 id="troubleshooting">Troubleshooting</h2>
<ul>
<li>If metawrap installation fail while using conda. A known fix is to install metawrap in a conda environment and put the path in "modules/metawrap_refine_bin.nf"
  To do so run the following command:</li>
</ul>
<pre><code class="language-sh">
#create an env and install metawrap
conda create -y -p /path/to/install/metawrap-env python=2.7
conda activate /path/to/install/metawrap-env
conda config --add channels defaults
conda config --add channels conda-forge
conda config --add channels bioconda
conda config --add channels ursky
conda install -y -c ursky metawrap-mg
conda deactivate

#edit MAFIN/modules/metawrap_refine_bin.nf to use the env of metawrap
#you need to change the line 3 and 25 to the path of your env (/path/to/install/metawrap-env)
</code></pre>
<ul>
<li>
<p>If you run the pipeline with google life sciences and get error code 14
  It means the process was killed by google, you just need to run the pipeline again don't forget to add "-resume"</p>
</li>
<li>
<p>If either Metawrap or checkm have the following error
  You need to increase the RAM in the command for local_engine and conda or in the "configs/containers.config"</p>
</li>
</ul>
<pre><code>IOError: [Errno 2] No such file or directory: 'binsA.checkm/storage/tree/concatenated.tre' 
</code></pre>
<ul>
<li>For other issue please open a ticket on <img alt="github" src="https://github.com/RVanDamme/MUFFIN/issues" /></li>
</ul>
<h2 id="options">Options</h2>
<h3 id="-cpus">--cpus</h3>
<ul>
<li>number of thread available</li>
<li>default 2</li>
</ul>
<h3 id="-mem">--mem</h3>
<ul>
<li>number of memory available</li>
<li>default 16g</li>
</ul>
<h3 id="-assembler">--assembler</h3>
<ul>
<li>which method to use</li>
<li>can be Hybrid with metaspades or long read + polishing with metaflye</li>
<li>required</li>
</ul>
<h3 id="-illumina">--illumina</h3>
<ul>
<li>location of the dir containing the forward and reverse illumina reads in fasta or fastq </li>
<li>required</li>
</ul>
<h3 id="-nanopore">--nanopore</h3>
<ul>
<li>location of the dir containing the nanopore reads in fasta or fastq</li>
<li>required</li>
</ul>
<h3 id="-output">--output</h3>
<ul>
<li>output directory</li>
<li>required</li>
</ul>
<h3 id="-profile">-profile</h3>
<ul>
<li>engine and executor</li>
<li>required</li>
<li>engine: local_engine, conda, docker, singularity</li>
<li>executor: local, gcloud, slurm</li>
</ul>
<h2 id="complete-help-and-options">Complete help and options</h2>
<pre><code>    *********hybrid assembly and differential binning workflow for metagenomics, transcriptomics and pathway analysis*********

    MUFFIN is still under development please wait until the first non edge version realease before using it.
    Please cite us using https://www.biorxiv.org/content/10.1101/2020.02.08.939843v1

    Mafin is composed of 3 part the assembly of potential metagenome assembled genomes (MAGs); the classification of the MAGs; and the annotation of the MAGs.

        Usage example:
    nextflow run RVanDamme/MUFFIN --output result --ont nanopore/ --illumina illumina/ --assembler metaspades --rna rna/ -profile local,docker
    or 
    nextflow run RVanDamme/MUFFIN --output result --ont nanopore/ --illumina illumina/ --assembler metaflye -profile local,docker

        Input:
    --ont                       path to the directory containing the nanopore read file (fastq) (default: $params.ont)
    --illumina                  path to the directory containing the illumina read file (fastq) (default: $params.illumina)
    --rna                       path to the directory containing the RNA-seq read file (fastq) (default: none)
    --bin_classify              path to the directory containing the bins files to classify (default: none)
    --bin_annotate              path to the directory containing the bins files to annotate (default: none)
    --assembler                 the assembler to use in the assembly step (default: $params.assembler)

        Optional input:
    --check_db                  path to the checkm database
    --check_tar_db              path to the checkm database tar compressed
    --sourmash_db               path to the LCA database for sourmash (default: GTDB LCA formated)
    --eggnog_db                 path to the eggNOG database

        Output:
    --output                    path to the output directory (default: $params.output)

        Outputed files:
        You can see the output structure at https://osf.io/a6hru/
    QC                          The reads file after qc
    Assembly                    The assembly contigs file 
    Bins                        The bins produced by CONCOCT, MetaBAT2, MaxBin2 and MetaWRAP (the refining of bins)
    Mapped bin reads            The fastq files containing the reads mapped to each metawrap bin
    Unmapped bin reads          The fastq files containing the unmmaped reads of illumina and nanopore
    Reassembly                  The reassembly files of the bins (.fa and .gfa)
    Checkm                      Various file outputed by CheckM (summary, taxonomy, plots and output dir)
    Sourmash                    The classification done by sourmash
    Classify summary            The summary of the classification and quality control of the bins (csv file)
    RNA output                  The de novo assembled transcript and the quantification by Salmon
    Annotation                  The annotations files from eggNOG (tsv format)
    Parsed output               HTML files that summarize the annotations and show graphically the pathways




        Basic Parameter:
    --cpus                      max cores for local use [default: $params.cpus]
    --memory                    80% of available RAM in GB for --metamaps [default: $params.memory]


        Workflow Options:
    --skip_ill_qc               skip quality control of illumina files
    --skip_ont_qc               skip quality control of nanopore file
    --short_qc                  minimum size of the reads to be kept (default: $params.short_qc )
    --filtlong                  use filtlong to improve the quality furthermore (default: false)
    --model                     the model medaka will use (default: $params.model)
    --polish_iteration          number of iteration of pilon in the polish step (default: $params.polish_iteration)
    --extra_ill                 a list of additional ill sample file (with full path with a * instead of _R1,2.fastq) to use for the binning in Metabat2 and concoct
    --extra_ont                 a list of additional ont sample file (with full path) to use for the binning in Metabat2 and concoct
    --skip_metabat2             skip the binning using metabat2 (advanced)
    --skip_maxbin2              skip the binning using maxbin2 (advanced)
    --skip_concoct              skip the binning using concoct (advanced)

        Nextflow options:
    -profile                    change the profile of nextflow both the engine and executor more details on github README
    -resume                     resume the workflow where it stopped
    -with-report rep.html       cpu / ram usage (may cause errors)
    -with-dag chart.html        generates a flowchart for the process tree
    -with-timeline time.html    timeline (may cause errors)
</code></pre>
<h2 id="bibliography">BIBLIOGRAPHY</h2>
<p>BWA: Li H. and Durbin R. (2009) Fast and accurate short read alignment with Burrows-Wheeler Transform. Bioinformatics, 25:1754-60. [PMID: 19451168] </p>
<p>CheckM: Parks DH, Imelfort M, Skennerton CT, Hugenholtz P, Tyson GW. 2015. CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes. Genome Research, 25: 1043–1055.</p>
<p>Concoct: Johannes Alneberg, Brynjar Smári Bjarnason, Ino de Bruijn, Melanie Schirmer, Joshua Quick, Umer Z Ijaz, Leo Lahti, Nicholas J Loman, Anders F Andersson &amp; Christopher Quince. 2014. Binning metagenomic contigs by coverage and composition. Nature Methods, doi: 10.1038/nmeth.3103 </p>
<p>Fastp: Shifu Chen, Yanqing Zhou, Yaru Chen, Jia Gu; fastp: an ultra-fast all-in-one FASTQ preprocessor, Bioinformatics, Volume 34, Issue 17, 1 September 2018, Pages i884–i890, https://doi.org/10.1093/bioinformatics/bty560</p>
<p>Filtlong: https://github.com/rrwick/Filtlong</p>
<p>Flye: Mikhail Kolmogorov, Jeffrey Yuan, Yu Lin and Pavel Pevzner, "Assembly of Long Error-Prone Reads Using Repeat Graphs", Nature Biotechnology, 2019 doi:10.1038/s41587-019-0072-8</p>
<p>HMMER: http://hmmer.org/ </p>
<p>Maxbin2: Wu YW, Tang YH, Tringe SG, Simmons BA, and Singer SW, "MaxBin: an automated binning method to recover individual genomes from metagenomes using an expectation-maximization algorithm", Microbiome, 2:26, 2014.</p>
<p>Medaka: https://github.com/nanoporetech/medaka</p>
<p>Metabat2: Kang DD, Froula J, Egan R, Wang Z. MetaBAT, an efficient tool for accurately reconstructing single genomes from complex microbial communities. PeerJ 2015;3:e1165. doi:10.7717/peerj.1165</p>
<p>Metawrap: Uritskiy, G.V., DiRuggiero, J. and Taylor, J. (2018). MetaWRAP—a flexible pipeline for genome-resolved metagenomic data analysis. Microbiome, 6(1). https://doi.org/10.1186/s40168-018-0541-1</p>
<p>Minimap2: Li, H. (2018). Minimap2: pairwise alignment for nucleotide sequences. Bioinformatics, 34:3094-3100. doi:10.1093/bioinformatics/bty191</p>
<p>Pilon: Bruce J. Walker, Thomas Abeel, Terrance Shea, Margaret Priest, Amr Abouelliel, Sharadha Sakthikumar, Christina A. Cuomo, Qiandong Zeng, Jennifer Wortman, Sarah K. Young, Ashlee M. Earl (2014) Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement. PLoS ONE 9(11): e112963. doi:10.1371/journal.pone.0112963</p>
<p>pplacer: Matsen FA, Kodner RB, Armbrust EV. 2010. pplacer: linear time maximum-likelihood and Bayesian phylogenetic placement of sequences onto a fixed reference tree. BMC Bioinformatics 11: doi:10.1186/1471-2105-11-538.</p>
<p>prodigal: Hyatt D, Locascio PF, Hauser LJ, Uberbacher EC. 2012. Gene and translation initiation site prediction in metagenomic sequences. Bioinformatics 28: 2223–2230.</p>
<p>Racon: Vaser R, Sovic I, Nagarajan N, Sikic M. 2017. Fast and accurate de novogenome assembly from long uncorrected reads. Genome Res 27:737–746.https://doi.org/10.1101/gr.214270.116</p>
<p>Samtools: Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R, and 1000 Genome Project Data Processing Subgroup, The Sequence alignment/map (SAM) format and SAMtools, Bioinformatics (2009) 25(16) 2078-9 [19505943]</p>
<p>Seqtk: https://github.com/lh3/seqtk</p>
<p>Sourmash: Brown et al, (2016), sourmash: a library for MinHash sketching of DNA, Journal of Open Source Software, 1(5), 27, doi:10.21105/joss.00027</p>
<p>Spades:  Lapidus A., Antipov D., Bankevich A., Gurevich A., Korobeynikov A., Nurk S., Prjibelski A., Safonova Y., Vasilinetc I., Pevzner P. A. New Frontiers of Genome Assembly with SPAdes 3.0.    (poster), 2014 </p>
<p>Unicycler: Wick RR, Judd LM, Gorrie CL, Holt KE (2017) Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads. PLoS Comput Biol 13(6): e1005595. https://doi.org/10.1371/journal.pcbi.1005595</p>
<h2 id="license">License</h2>
<p>Code is <a href="LICENSE">GPL-3.0</a></p>
<h2 id="contributing">Contributing</h2>
<p>We welcome contributions from the community! See our
<a href="CONTRIBUTING.md">Contributing</a> guidelines</p>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="assets/javascripts/vendor.7e0ee788.min.js"></script>
      <script src="assets/javascripts/bundle.b3a72adc.min.js"></script><script id="__lang" type="application/json">{"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}</script>
      
      <script>
        app = initialize({
          base: ".",
          features: ['navigation.tabs'],
          search: Object.assign({
            worker: "assets/javascripts/worker/search.4ac00218.min.js"
          }, typeof search !== "undefined" && search)
        })
      </script>
      
    
  </body>
</html>